ssr/src/page/wallet/tokens.rs

```diff
@@ -33,8 +30,15 @@ impl KeyedData for Principal {
         *self
     }
 }
+impl KeyedData for String {
+    type Key = String;
+
+    fn key(&self) -> Self::Key {
+        self.clone()
+    }
+}
 impl CursoredDataProvider for TokenRootList {
-    type Data = Principal;
+    type Data = String;
```

Comment:
Strong typing!!

Reply:
Yeah the reason why its a string its because the param could be a hardcoded type like "btc" and "usdc" which will be changed to ledger instead of root so it should be fixed...